protostuff-collectionschema/src/main/java/io/protostuff/MapSchema.java

```diff
@@ -47,6 +47,44 @@
         public Class<?> typeClass();
     }
 
+    private static final Comparable EMPTY_OBJECT = new Comparable()
+    {
+        @Override
+        public int compareTo(Object o)
+        {
+            return o == this ? 0 : -1;
+        }
+        @Override
+        public boolean equals(Object o)
+        {
+            return o == this;
+        }
+        @Override
+        public int hashCode()
+        {
+            return 0;
+        }
+    };
+
+    @SuppressWarnings("unchecked")
+    public static <T extends Map> T removeUnmodifiableWrapper(T map, MessageFactory messageFactory)
+    {
+        if(map == null)
+            return null;
+
+        try
+        {
+           map.put(EMPTY_OBJECT, EMPTY_OBJECT);
+           map.remove(EMPTY_OBJECT);
+        }

```

Comment:
Not in favor of this hack.
Surely there is a solution to know if a collection is unmodifable without adding/removing?